Rubberduck.Core/UI/Refactorings/IRefactoringDialogFactory.cs

```diff
@@ -0,0 +1,18 @@
+﻿using Rubberduck.Refactorings;
+
+namespace Rubberduck.UI.Refactorings
+{
+    public interface IRefactoringDialogFactory
+    {
+        TDialog CreateDialog<TModel, TView, TViewModel, TDialog>(TModel model, TViewModel viewmodel)
+            where TModel : class
+            where TView : class, IRefactoringView, new()
+            where TViewModel : class, IRefactoringViewModel<TModel>
+            where TDialog : class, IRefactoringDialog<TModel, TView, TViewModel>;
+        void ReleaseDialog(IRefactoringDialog dialog);
+        TViewModel CreateViewModel<TModel, TViewModel>(TModel model)
+            where TModel : class
+            where TViewModel : class, IRefactoringViewModel<TModel>;
+        void ReleaseViewModel(IRefactoringViewModel viewModel);
```

Comment:
and here either?

Reply:
Same situation; `TViewModel` must be `IRefactoringViewModel<>`, which derives from non-generic `IRefactoringViewModel`. [See here](https://github.com/rubberduck-vba/Rubberduck/pull/4072/files/7cd7fd05b902790dfea0835a05b7b95a37bd8418#diff-b7c129facafd1c1197cf0cd609932786R5).